modules/header/react/lib/parts/SearchBar.tsx

```diff
@@ -0,0 +1,245 @@
+  {
+    position: 'relative',
+    flexGrow: 1,
+    display: 'flex',
+    alignItems: 'center',
+  },
+  ({ rightAlign, isCollapsed, grow }) => {
+    const rightAlignStyles: CSSObject = rightAlign
+      ? {
+        display: 'flex',
+        maxWidth: '480px',
+      }
+      : {};
+    const collapseStyles: CSSObject = isCollapsed
+      ? {
+        top: 0,
+        right: 0,
+        left: 0,
+        bottom: 0,
+        margin: 0,
+        position: 'absolute',
+        background: colors.frenchVanilla100,
+        maxWidth: 'unset',
+        zIndex: 1,
+        '&.search-enter': {
+          opacity: 0,
+        },
+        '&.search-enter-active': {
+          opacity: 1,
+          transition: `opacity ${mobileTransitionDuration}ms`,
+        },
+        '&.search-exit': {
+          opacity: 1,
+        },
+        '&.search-exit-active': {
+          opacity: 0,
+          transition: `opacity ${mobileTransitionDuration}ms`,
+        },
+      }
+      : {};
+    const growStyles: CSSObject = grow
+      ? {
+        width: '100%',
+        maxWidth: '100%',
+      }
+      : {};
+    return { ...rightAlignStyles, ...collapseStyles, ...growStyles };
+  },
+);
+
+const inputHeight = spacing.l
+const SearchSubmit = styled(IconButton)({
+  width: inputHeight,
+  height: inputHeight,
+  position: `absolute`,
+  margin: `${spacing.xxxs} ${spacing.xxs}`,
+  bottom: `${spacing.s}`,
+  zIndex: 1,
+});
+
+const iconButtonType = (shouldInverse: boolean) => (
+  shouldInverse ? IconButton.Types.Inverse : IconButton.Types.Plain
+);
+
+export class SearchBar extends React.Component<SearchBarProps, SearchBarState> {
+  static defaultProps = {
+    themeColor: HeaderTheme.White,
+    headerHeight: HeaderHeight.Small,
+    placeholder: 'Search',
+    accessibleLabel: 'Search',
+    showClearButton: true,
+  };
+
+  private inputRef = React.createRef<HTMLInputElement>();
+  private formRef = React.createRef<HTMLFormElement>();
+
+  state: Readonly<SearchBarState> = {
+    mobileToggle: false,
+    searchQuery: '',
+    isFocused: false,
+  };
+
+  handleSubmit = (event: React.FormEvent<HTMLFormElement>): void => {
+    event.preventDefault();
+    if (this.state.searchQuery.trim()) {
+      this.props.onSubmit(event);
+    } else {
+      this.focusInput();
+    }
+  }
+
+  openMobileSearch(): void {
+    this.setState({ mobileToggle: true });
+  }
+
+  closeMobileSearch(): void {
+    this.setState({ mobileToggle: false });
+  }
+
+  focusInput(): void {
+    if (this.inputRef.current) {
+      this.inputRef.current.focus();
+    }
+  }
+
+  handleSearchInputChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+    event.preventDefault();
+    this.setState({ searchQuery: event.target.value });
+    if (this.props.onInputChange) {
+      this.props.onInputChange(event)
+    }
+  };
+
+  clearForm = (): void => {
+    if (this.formRef.current) {
+      this.formRef.current.reset()
+    }
+  }
+
+  clearButtonType = iconButtonType(
```

Comment:
might be more readable to have this method take in whether its a clear/submit button and evaluate the type based on that